pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -167,32 +119,110 @@ func (a *Actuator) Create(ctx context.Context, cluster *clusterv1.Cluster, machi
 	}
 
 	controlPlaneMachines := GetControlPlaneMachines(clusterMachines)
+	if len(controlPlaneMachines) == 0 {
+		return errors.New("unable to proceed with machine creation until there is at least 1 control plane machine")
+	}
 
-	isNodeJoin, err := a.isNodeJoin(scope, controlPlaneMachines)
-	if err != nil {
-		return errors.Wrapf(err, "failed to determine whether machine %q should join cluster %q", machine.Name, cluster.Name)
+	/*
+	If new machine is a control plane machine, we may be in the following states:
```

Comment:
This is almost like a lease, I haven't thought it all the way through but I'm wondering if we can reuse v1coordination.Lease

Reply:
That lease stuff looks really cool! I think there's a fairly similar shape here for sure.

I'd still be concerned about pivoting a lease, and I'm not sure how to set the timeout – I think it's preferable to me to stop doing anything than it is to end up with a split brain cluster.

Reply:
Re: Lease - a Slack conversation between myself and @ncdc:

> Andrew Kutz  [3 hours ago]
> FWIW @andygoldstein, `coordination.Lease` is `v1beta1` in 1.13 and GA in 1.14. It’s likely fine to use that since you can support 1.13 with the beta type.
> 
> Andy Goldstein  [3 hours ago]
> unless @detiberusj you think we can use this?
> 
> REMOVED
> 
> Andrew Kutz  [3 hours ago]
> But still, maybe you can figure out how to use the `Lease`
> 
> Andrew Kutz  [3 hours ago]
> The only examples I can find are for leader election
> 
> Andy Goldstein  [3 hours ago]
> there’s a package for it, 1 sec
> 
> Andrew Kutz  [3 hours ago]
> It doesn’t seem much more than a formalized version of what we’re doing. They just make things first-class values instead of shoving an encoded struct into a field
> 
> Andrew Kutz  [3 hours ago]
> Yeah, I dug into that pretty deep last weekend
> 
> Andrew Kutz  [3 hours ago]
> https://godoc.org/k8s.io/kubernetes/pkg/apis/coordination
> 
> Andrew Kutz  [3 hours ago]
> Again, they do have a `v1beta1` type as well that can support 1.13
> 
> Andrew Kutz  [3 hours ago]
> I like the config object because it’s not tied to the K8s version
> 
> Andrew Kutz  [3 hours ago]
> And it’s super simple to do
> 
> Andy Goldstein  [3 hours ago]
> yeah, i think that’s best (configmap). i’ll update so the cluster creates it
> 
> Andy Goldstein  [3 hours ago]
> and machines gate on it
> 
> Andrew Kutz  [3 hours ago]
> Anyway, the LeaseSpec doesn’t seem to provide the framework for using the lease. It just provides a first-class schema for defining a lease holder
> 
> Andy Goldstein  [3 hours ago]
> yea
> 
> Andrew Kutz  [3 hours ago]
> I wanted the actual framework for using the lease.
> 
> Andy Goldstein  [3 hours ago]
> https://github.com/kubernetes/client-go/blob/master/tools/leaderelection/resourcelock/interface.go
> tools/leaderelection/resourcelock/interface.go
> 
> Andy Goldstein  [3 hours ago]
> that?
> 
> Andrew Kutz  [3 hours ago]
> Yeah, but that’s tied to leader election, and I guess that’s kind of what an exclusive lock is? But I was hoping for an implementation of `sync.Locker` to keep things super simple
> 
> Andy Goldstein  [3 hours ago]
> i thought that was a generic locking interface?
> 
> Andrew Kutz  [3 hours ago]
> `sync.Locker`? It is
> 
> 
> Andrew Kutz  [3 hours ago]
> But the linked leader election package doesn’t do anything I see that maps to that. It’s more of a means of managing the leases?
> 
> Andy Goldstein  [3 hours ago]
> oh ok i see what you’re saying
>
> Andrew Kutz  [3 hours ago]
Basically it’s not much more than what we’re doing with config maps, just tied to the leases
>
> Andrew Kutz  [3 hours ago]
> It’s still just CRUD based